### PR TITLE
Dml regview fix 2

### PR DESCRIPTION
**Pull request by JonatanWaern**
Previously https://github.com/intel/device-modeling-language/pull/8

### CODE REVIEW

File: lib/1.4/dml-builtins.dml [hunk 1]
```diff
@@ -1359,28 +1359,43 @@ template bank is (object, shown_desc, _qname) {
...
+            local uint64 num = 0;
+            foreach r in (this._each_register) {
+                if (r.offset != unmapped_offset) {
+                    _reginfo_cache[num] = r;
+                    num++;
                 }
                 num++;
             }
         }
         throw;
```
Comment (mandolaerik) [inline lib/1.4/dml-builtins.dml:1384]: Gives compile error on my side. Please run tests.

File: lib/1.4/dml-builtins.dml [hunk 2]
```diff
@@ -1359,28 +1359,43 @@ template bank is (object, shown_desc, _qname) {
...
 
-    // would be great if we could index a sequence
-    shared method _get_register(uint32 reg) -> (register) throws default {
-        local uint32 num = 0;
-        foreach r in (this._each_register) {
-            if (r.offset != unmapped_offset) {
-                if (num == reg) {
-                    return r;
+
+
```
Comment (mandolaerik) [inline lib/1.4/dml-builtins.dml:1363]: excess blank lines

File: lib/1.4/dml-builtins.dml [hunk 3]
```diff
@@ -1359,28 +1359,43 @@ template bank is (object, shown_desc, _qname) {
...
+                }
+            }
+            _reginfo_cache = new register[_reginfo_cache_size];
+            local uint64 num = 0;
+            foreach r in (this._each_register) {
+                if (r.offset != unmapped_offset) {
+                    _reginfo_cache[num] = r;
+                    num++;
                 }
                 num++;
```
Comment (mandolaerik) [inline lib/1.4/dml-builtins.dml:1381]: This line looks wrong.

I'd appreciate an `assert n==_reginfo_cache_size` in the end.

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9581588

**Review by mandolaerik — APPROVED** [review]

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9581587

**Pull request merged**